packages/langchain/src/openai/util.ts

```diff
@@ -36,39 +38,49 @@ import type {
 /**
  * Maps a LangChain {@link BindToolsInput} to {@link AzureOpenAiChatCompletionFunctions}.
  * @param tool - Base class for tools that accept input of any shape defined by a Zod schema.
+ * @param strict - Whether to enforce strict mode for the function call.
  * @returns The OpenAI chat completion function.
  * @internal
  */
 export function mapToolToOpenAiFunction(
-  tool: BindToolsInput
+  tool: ChatAzureOpenAIToolType,
+  strict?: boolean
 ): AzureOpenAiFunctionObject {
-  if (isToolDefinition(tool)) {
+  if (isToolDefinitionLike(tool)) {
     return {
       name: tool.function.name,
       description: tool.function.description,
-      parameters: tool.function.parameters ?? { type: 'object', properties: {} }
+      parameters: tool.function.parameters ?? {
+        type: 'object',
+        properties: {}
+      },
+      ...(strict !== undefined ? { strict } : {})
     };
   }
   return {
     name: tool.name,
     description: tool.description,
     parameters: isZodSchema(tool.schema)
       ? zodToJsonSchema(tool.schema)
-      : tool.schema
+      : tool.schema,
+    ...(strict !== undefined ? { strict } : {})
   };
 }
 
 /**
  * Maps a LangChain {@link BindToolsInput} to {@link AzureOpenAiChatCompletionTool}.
```

Comment:
Maps `{@link ChatAzureOpenAIToolType}` instead of `BindToolsInput`

Reply:
Fixed also other occurrences.